src/master-playlist-controller.js

```diff
@@ -1265,6 +1263,28 @@ export class MasterPlaylistController extends videojs.EventTarget {
       }
     }
 
+    // fmp4 relies on browser support, while ts relies on muxer support
+    const supportFunction = mainStartingMedia.isFmp4 ? browserSupportsCodec : muxerSupportsCodec;
+    const audioSupported = !codecs.hasOwnProperty('audio') ? true : supportFunction(codecs.audio);
```

Comment:
This part of the code works a bit weird. We don't want to check if we support a codec if we don't have one in the `codecs` object. If we do have it we use the return value of `supports` to see if it can actually be played. The suggested change would mean that any codec existence would mean that we support it.

Reply:
Ah, yeah, it's not equivalent but then I'm still a bit confused on what the original code does has audio not being present would make `audioSupported` set to `true`.

```js
var o = { audio: 1}
> !o.hasOwnProperty('audio') ? true : 1
< 1
> o.hasOwnProperty('audio') || 1
< true
> !o.hasOwnProperty('video') ? true : 1
< true
> o.hasOwnProperty('video') || 1
< 1
```

Reply:
Yeah `audioSupportedOrNoAudio` would have been a better name